web/web.go

```diff
@@ -0,0 +1,799 @@
+// 	if err != nil {
+// 		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), 500)
+// 		return
+// 	}
+// 	var t setT
+// 	err = json.Unmarshal(reqBody, &t)
+// 	if err != nil {
+// 		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), 400)
+// 		return
+// 	}
+// 	i := 1
+// 	for k, v := range t {
+// 		//fmt.Printf("%d: key (`%T`)`%v`, value (`%T`)`%#v`\n", i, k, k, v, v)
+// 		//fmt.Println(mysql.ToSpannerType[k].T.Name)
+// 		mysql.ToSpannerType[k].T.Name = v
+// 		i++
+// 	}
+// 	// for _, spTable := range conv.SpSchema {
+// 	// 	for _, colDef := range spTable.ColDefs {
+// 	// 		colDef.T.Name = mysql.ToSpannerType[k].T.Name
+// 	// 	}
+// 	// }
+// 	w.WriteHeader(http.StatusOK)
+// 	//json.NewEncoder(w).Encode(editTypeMap)
+// }
+
+func setTypeMapGlobal(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), 500)
+		return
+	}
+	var t setT
+	err = json.Unmarshal(reqBody, &t)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), 400)
+		return
+	}
+
+	for k, v := range app.conv.SpSchema {
+		for kk, _ := range v.ColDefs {
+			for tk, tv := range t {
+				sourceTable := app.conv.ToSource[k].Name
+				sourceCol := app.conv.ToSource[k].Cols[kk]
+				srcCol := app.conv.SrcSchema[sourceTable].ColDefs[sourceCol]
+				if srcCol.Type.Name == tk {
+					var ty ddl.Type
+					var issues []internal.SchemaIssue
+					switch app.driver {
+					case "mysql", "mysqldump":
+						ty, issues = toSpannerTypeMySQL(app.conv, srcCol.Type.Name, tv, srcCol.Type.Mods)
+					case "pg_dump", "postgres":
+						ty, issues = toSpannerTypePostgres(app.conv, srcCol.Type.Name, tv, srcCol.Type.Mods)
+					default:
+						http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), 400)
+						return
+					}
+
+					if len(srcCol.Type.ArrayBounds) > 1 {
+						ty = ddl.Type{Name: ddl.String, Len: ddl.MaxLength}
+						issues = append(issues, internal.MultiDimensionalArray)
+					}
+					if srcCol.Ignored.Default {
+						issues = append(issues, internal.DefaultValue)
+					}
+					if srcCol.Ignored.AutoIncrement {
+						issues = append(issues, internal.AutoIncrement)
+					}
+					if len(issues) > 0 {
+						app.conv.Issues[sourceTable][srcCol.Name] = issues
+					}
+					ty.IsArray = len(srcCol.Type.ArrayBounds) == 1
+					tempSpSchema := app.conv.SpSchema[k]
+					tempColDef := tempSpSchema.ColDefs[kk]
+					tempColDef.T = ty
+					tempSpSchema.ColDefs[kk] = tempColDef
+					app.conv.SpSchema[k] = tempSpSchema
+				}
+			}
+		}
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+func remove(slice []string, s int) []string {
+	return append(slice[:s], slice[s+1:]...)
+}
+func removePk(slice []ddl.IndexKey, s int) []ddl.IndexKey {
+	return append(slice[:s], slice[s+1:]...)
+}
+func setTypeMapTableLevel(w http.ResponseWriter, r *http.Request) {
```

Comment:
Usual go style guidelines are to keep functions to about a page (about 50 lines or so)
This is much longer -- are there good ways to split this up into a couple of functions?

Reply:
updated function, splitted into multiple smaller functions